### ISSUE TITLE
[Request]: Launcher cannot locate Genshin when installed through Google Play

### ISSUE THREAD

**Issue by swhong3517** [CollapseLauncher/Collapse#541]
### Is your request related to a problem?

Yes

### Describe your proposed solution

Hi. Just got back into Genshin, installed the game through Google Play PC beta.
But when I press "Install/Locate Game" Collapse doesn't seem to recognize the installed version and tries to do a full install instead. Google Play worked on Collapse when I played Genshin last year, so I'm confused on whether if it's a bug or Collapse just doesn't  support games downloaded from Google Play anymore.

### Describe alternatives you've considered

(None)

**Comment by bagusnl** [comment]
can you send us the link to the launcher for the Google Play PC Beta and also the app.info file inside GenshinImpact_Data folder?

**Comment by swhong3517** [comment]
By "launcher", do you mean the link to the Google Play beta download site?
Here's the INFO file: https://gofile.io/d/v5AxlN

**Comment by bagusnl** [comment]
is this the same as this one https://github.com/CollapseLauncher/Collapse/issues/474 ?

**Comment by bagusnl** [comment]
The app.info file seems correct for Genshin Global region. So as long as the GenshinImpact.exe file is not modified then it should be fine technically

**Comment by swhong3517** [comment]
Should I try and reinstall Collapse then? If I uninstall, will it affect games that are directly installed via Collapse?

**Comment by bagusnl** [comment]
I dont think it will help, try to move config.ini file to different folder as a backup and then locate your installation again in collapse. While at it, send the file here too

**Comment by swhong3517** [comment]
Here it is: https://gofile.io/d/71TqtP

**Comment by swhong3517** [comment]
Moving the config file doesn't work either.

**Comment by bagusnl** [comment]
Investigating rn... 
Google Play beta should have the same config as Global region but idk why it cant detect it

**Comment by bagusnl** [comment]
Can you try this build https://github.com/CollapseLauncher/Collapse/actions/runs/10042690527 (when its complete, just download the artifact, extract, and run CollapseLauncher.exe)

Open Settings > Open App Data Folder
go to `_metadatav3` folder and add/replace these 2 files

[config_GIGlbGPlay.json](https://github.com/user-attachments/files/16335235/config_GIGlbGPlay.json)
[stamp.json](https://github.com/user-attachments/files/16335236/stamp.json)

**MAKE SURE** to go to Settings > Clear Metadata and Restart **BEFORE** opening release version of Collapse

**Comment by swhong3517** [comment]
![001](https://github.com/user-attachments/assets/0c44458d-47b3-46c7-a9cc-a299c22981a6)
![002](https://github.com/user-attachments/assets/038a1d7c-4fe3-43f6-93b4-184b4d9d8e75)
![003](https://github.com/user-attachments/assets/f15bfcd4-a94e-47cd-99a1-84a710623da6)
Did as instructed, while the launcher now shows a separate region for Google Play, it still doesn't detect the installed version.
Should I just uninstall Genshin and try downloading directly from the launcher?

**Comment by bagusnl** [comment]
Enable console in the settings and try to locate again. See what the message says

**Comment by swhong3517** [comment]
[Warn]    [InvokeProp::PreventSleep()] Starting to prevent sleep!
[Info]    [TryFindGamePathFromExecutableAndConfig] Fail!
[Info]    Gathering packages information for installation (State: NotInstalled)...
[Info]    Initializing Sophon Chunk download method with Main Thread: 4, Chunks Thread: 4 and Max HTTP handle: 16

**Comment by bagusnl** [comment]
I will take a look tomorrow while renting a server. My internet is not fast enough to download the client

**Comment by studiobuttermedia** [comment]
Leaving it here, config.ini:
```
channel=1
sub_channel=6
cps=pcgoogle
```

**Comment by studiobuttermedia** [comment]
@swhong3517 Let me explain the situation, the one you have the Google Play PC one has a different config.ini file which currently does not work with Collapse. Collapse currently works with the game downloaded from HoYoverse official servers. You could just change the config.ini to the following:
```
channel=1
sub_channel=0
cps=hoyoverse
```
And do a game detection by going to the region "Global", choose find games, locate where `GenshinImpact.exe` is and run the game there. Nothing much gonna change except when you want to purchase Welkin or Crystals, you will be greeted with HoYoverse's Payment system instead of Google Play

**Comment by bagusnl** [comment]
![image](https://github.com/user-attachments/assets/808babec-ed4f-476a-8ae9-fa6b75cd600c)

Can you check the directory in which you point Collapse too?
I have tried fully installing Genshin using Gplay version of HoYoPlay and it detects the game just fine

**Comment by bagusnl** [comment]
I'm not sure since HI3 is quite more messy (too much region)

I'll talk to the dev team on implementing a payment system switcher instead to minimize adding too much unnecessary regions

**Comment by bagusnl** [comment]
Fixed in 1.81.1 https://github.com/CollapseLauncher/Collapse/releases/tag/CL-v1.81.1-pre
If you still have problem, reopen this ticket or open another one.

Thanks for the support <3
